neqo-qpack/src/qpack_helper.rs

```diff
@@ -43,7 +82,41 @@ impl<'a> BufWrapper<'a> {
         }
     }
 
-    pub fn slice(&mut self, len: usize) -> Res<&[u8]> {
+    pub fn done(&self) -> bool {
+        self.offset == self.buf.len()
+    }
+
+    pub fn read_prefixed_int(&mut self, prefix_len: u8) -> Res<u64> {
```

Comment:
It might pay to document these, including the assumptions you are making.  That is, that the underlying buffer is complete.